integrations/slack/src/actions/retreive-message.ts

```diff
@@ -0,0 +1,34 @@
+import { WebClient } from '@slack/web-api'
+import { Integration } from '.botpress'
+
+export const retrieveMessage: Integration['actions']['retrieveMessage'] = async ({ ctx, input }) => {
+  const slackClient = new WebClient(ctx.configuration.botToken)
+
+  const response = await slackClient.conversations.history({
+    limit: 1,
+    inclusive: true,
+    latest: input.ts,
+    channel: input.channel,
+  })
+
+  if (response.ok) {
+    throw new Error(`Could not retrieve message: ${response.error}`)
```

Comment:
**suggestion:** maybe you can also log the message to the bot using `logger.forBot()` or event better, we could add this in a `try {} catch {}` statement in the SDK